exonum-java-binding-core/rust/ejb-app/src/node_builder.rs

```diff
@@ -0,0 +1,144 @@
+    let mut service_factories = HashMap::new();
+    service_factories.insert(
+        CONFIGURATION_SERVICE.to_owned(),
+        Box::new(ConfigurationServiceFactory) as Box<ServiceFactory>,
+    );
+    service_factories.insert(
+        BTC_ANCHORING_SERVICE.to_owned(),
+        Box::new(BtcAnchoringServiceFactory) as Box<ServiceFactory>,
+    );
+    service_factories.insert(
+        EJB_SERVICE.to_owned(),
+        Box::new(JavaServiceFactory) as Box<ServiceFactory>,
+    );
+    service_factories
+}
+
+#[doc(hidden)]
+pub fn services_to_enable<P: AsRef<Path>>(path: P) -> HashSet<String> {
+    let ServicesToEnable { mut services } = ConfigFile::load(path).unwrap_or(ServicesToEnable {
+        services: {
+            let mut services = HashSet::new();
+            services.insert(CONFIGURATION_SERVICE.to_owned());
+            services
+        },
+    });
+
+    // Add EJB_SERVICE if it's missing
+    services.insert(EJB_SERVICE.to_owned());
+
+    services
+}
+
+pub fn create() -> fabric::NodeBuilder {
+    let services = services_to_enable(PATH_TO_SERVICES_TO_ENABLE);
+    let mut service_factories = service_factories();
+
+    let mut builder = fabric::NodeBuilder::new();
+    for service_name in &services {
+        match service_factories.remove(service_name) {
+            Some(factory) => {
+                builder = builder.with_service(factory);
+            }
+            None => panic!("Found unknown service name {}", service_name),
+        }
+    }
+    builder
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::io::Write;
+    use tempfile::{Builder, TempPath};
+
+    fn create_config(filename: &str, cfg: &str) -> TempPath {
+        let mut cfg_file = Builder::new().prefix(filename).tempfile().unwrap();
+        writeln!(cfg_file, "{}", cfg).unwrap();
+        cfg_file.into_temp_path()
+    }
+
+    #[test]
+    fn no_config() {
+        let services_to_enable = services_to_enable("");
+        assert_eq!(services_to_enable.len(), 2);
+        assert!(services_to_enable.contains(EJB_SERVICE));
+        assert!(services_to_enable.contains(CONFIGURATION_SERVICE));
+    }
+
+    #[test]
+    fn empty_list() {
+        let cfg = create_config("empty_list.toml", "services = []");
+        let services_to_enable = services_to_enable(cfg);
+        assert_eq!(services_to_enable.len(), 1);
+        assert!(services_to_enable.contains(EJB_SERVICE));
+    }
+
+    #[test]
+    fn duplicated() {
+        let cfg = create_config(
+            "duplicated.toml",
+            "services = [\"btc-anchoring\", \"btc-anchoring\"]",
+        );
+        let services_to_enable = services_to_enable(cfg);
+        assert_eq!(services_to_enable.len(), 2);
+        assert!(services_to_enable.contains(EJB_SERVICE));
+        assert!(services_to_enable.contains(BTC_ANCHORING_SERVICE));
+    }
+
+    #[test]
+    fn broken_config() {
+        let cfg = create_config("broken.toml", "not_list = 1");
```

Comment:
Shan't we throw in this case?

Reply:
Hmmm... No, be just use default config... I'll consider panicing